src/app/shared/components/typeahead/typeahead.component.ts

```diff
@@ -28,13 +28,13 @@ import { OperatorFunction } from 'rxjs/internal/types';
 })
 export class TypeaheadComponent implements OnInit, OnDestroy {
   @Input() control: UntypedFormControl = new UntypedFormControl();
-  @Input() set data (value: any[]) {
-    if (this.control){
+  @Input() set data(value: any[]) {
+    if (this.control) {
       this.setControlValue(null);
     }
     this._data.next(value);
-    if (this.control){
-      this.setControlValue(this.data[0]);
+    if (this.control) {
+      this.setControlValue(this.data ? this.data[0] : null);
```

Comment:
Shouldn't this be testing for length > 0?

Reply:
doing a check where length > 0 would still throw an error because the issue is that this.data can be undefined.